<a href="https://colab.research.google.com/github/hamiq/parkinsons_prediction/blob/main/Parkinson's_Detection.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
import os, sys
from sklearn.preprocessing import MinMaxScaler
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

In [2]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/parkinsons.data')
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
0  phon_R01_S01_1      119.992       157.302  ...  0.266482  2.301442  0.284654
1  phon_R01_S01_2      122.400       148.650  ...  0.335590  2.486855  0.368674
2  phon_R01_S01_3      116.682       131.111  ...  0.311173  2.342259  0.332634
3  phon_R01_S01_4      116.676       137.871  ...  0.334147  2.405554  0.368975
4  phon_R01_S01_5      116.014       141.781  ...  0.234513  2.332180  0.410335

[5 rows x 24 columns]

In [3]:
# get features and labels
# features are all columns except status
# labels are those in the status column and have values 0 and 1

features = df.loc[:, df.columns != 'status'].values[:, 1:]
labels = df.loc[:, 'status'].values

In [7]:
# get count of each label

print('Count of 0: ', labels[labels==0].shape[0],'\nCount of 1: ', labels[labels==1].shape[0])

Count of 0:  48 
Count of 1:  147


In [8]:
# scale the features to between -1 and 1
# assign features to x (independent) and labels to y (dependent)

scaler = MinMaxScaler((-1,1))
x = scaler.fit_transform(features)
y = labels

Feature scaling is a method used to normalize the range of independent variables (or features of data). For example, if you have multiple independent values like age, salary, and height with their range as 18-100 years, 60,000-120,000 USD, and 4-7 feet, feature scaling would help them all to be in the same range.

The importance of feature scaling is that it allows the model to compare each features attributes against each other without one feature impacting the model singificantly due to its larger size. For example if two features are weight and price, we know that these two features do not have a meaningful comparison but the machine would make the assumption that weight is more important than price because of the larger values, so scaling would have to be done in the preprocessing step.

In [9]:
# split dataset into training and testing sets

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=7)

In [10]:
# train the model

model = XGBClassifier()
model.fit(x_train, y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

In [11]:
# test the model by calculating its accuracy

y_pred = model.predict(x_test)
print(accuracy_score(y_test, y_pred)*100)

94.87179487179486


The model predicted the onset of Parkinson's disease in the test set with an accuracy of 94.87%.